### DISCUSSION TITLE
[Bug]: iCloud Drive not downloaded files you can't access from Nimble

### DISCUSSION THREAD

**Discussion by piratx** [mikekazakov/nimble-commander#388] [Q&A] [answered]
### Confirmation

- [X] I have read the contributing guidelines

### Version of Nimble Commander

Version 1.6.1 (4088), MacOS 14.5 (23F79

### Distribution Type

Downloaded directly from the website (including Brew)

### Expected Result

<img width="1041" alt="Screenshot 2024-09-20 at 7 31 33 AM" src="https://github.com/user-attachments/assets/f4c4a953-a36b-4693-a109-50fd863da2c6">

In the above screenshot, at the top we see some files I created earlier this week in iCloud Drive.  What's "missing" from Nimble Commander to have me able to open those files, is a Cloud download button as seen in the bottom part of this screenshot.

Will Nimble Commander could somehow to act the same as Finder?
You should double click one of the files to open which it works but in large files it might take 1 minute or so
or to hit spacebar (quicklook) on of the items to open, which never works :(
<img width="798" alt="Screenshot 2024-09-20 at 7 43 06 AM" src="https://github.com/user-attachments/assets/16bfc056-be8c-4985-9998-af18ee6b3f54">

In Finder also you have a right click > Download Now option which is really nice if you are in an old folder with many files and you want to Download all this with a click.

### Actual Result

In Nimble Commander you hit the spacebar and just a simple quicklook with the pdf file icon can be seen, the only way to open it is double click and wait.
If you first open the file in Preview eg, you should be able to spacebar it later just fine.

### Steps to reproduce

Go into your iCloud Drive files, a folder that in mac have those cloud icons with the down arrow below as seen on the screenshot above. Those files shouldn't be accessible on Nimble if you dont open them first

### Additional Information

It would be nice to have a column like the new one for tags on Nimble Commander! :)

**Comment by piratx** [comment]
This is not a bug actually.. sorry about that! There might be a more suitable tag.

**Comment by mikekazakov** [comment]
Moving into Q&A section of Discussions.

**Comment by mikekazakov** [comment, marked as answer]
Here's a snippet from the F.A.Q. section of Help.md (https://github.com/mikekazakov/nimble-commander/blob/main/Docs/Help.md#frequently-asked-questions), written many years ago:

> Q: Can Nimble Commander access iCloud storage?
A: Nimble Commander does not provide first-class access to iCloud. Apple does not have an official API to allow applications to directly manipulate items outside of their own containers. At least that’s my understanding of the status quo at the moment of writing (I’d be glad to be proven wrong). However, you can manually navigate to `~/Library/Mobile Documents/com~apple~CloudDocs` and access the items there using the normal UI of Nimble Commander. This usually works, but there is no guarantee that the content of that folder is properly synchronized.

> Will Nimble Commander could somehow to act the same as Finder?

I'm not aware of any official API that provides this functionality, if anybody has any hints - welcome to share them.
So for now the answer is unfortunately "no".
